lib/bitcoin/cw_bitcoin.dart

```diff
@@ -802,18 +802,22 @@ class CWBitcoin extends Bitcoin {
   }
 
   @override
-  String? getTransactionAddress(Object wallet, TransactionInfo tx) {
+  List<String>? getTransactionAddresses(Object wallet, TransactionInfo tx) {
     final bitcoinWallet = wallet as BitcoinWallet;
     final bitcoinTx = tx as ElectrumTransactionInfo;
 
     if (bitcoinTx.unspents == null || bitcoinTx.unspents!.isEmpty) {
       return null;
     }
 
-    // final bitcoinSPAddrs =
-    //     bitcoinTx.unspents!.first.bitcoinAddressRecord as BitcoinSilentPaymentAddressRecord;
-    final bitcoinSPAddrs = wallet.walletAddresses as BitcoinWalletAddresses;
-    final bitcoinSPAddr = bitcoinSPAddrs.silentAddresses.first;
-    return bitcoinSPAddr.address;
+    final addresses = <String>[];
+    final labels = <String>[];
+    bitcoinTx.unspents!.forEach((unspent) {
+      addresses.add(bitcoinWallet.walletAddresses.silentAddresses
+          .firstWhere((address) => address.silentPaymentTweak == unspent.silentPaymentLabel)
+          .address);
+    });
```

Comment:
wrap this with a try and catch, as this will be probably called with also normal transactions, so I believe `firstWhere` would throw an exception at that time

also rename the function to `getSPTransactionAddresses` since it only relates to SP